lib/runner/extract-runnable-items.js

```diff
@@ -186,9 +188,14 @@ var sdk = require('postman-collection'),
 
         items = findItemsOrGroups(collection, entrypointLookup, true);
 
-        // Sorting the items by the order in which they appear as folder arguments.
-        items && items.members && items.members.sort(function (a, b) {
-            return entrypoints.indexOf(a.name) - entrypoints.indexOf(b.name);
+        // Sorting the items and folders by the order in which they appear as folder/request arguments,
+        // only if specified in the entrypoint.preserveOrder
+        preserveOrder && items && items.members && _.isArray(items.members) && items.members.sort(function (a, b) {
+            // Getting the keys of the folder/request as lookups can be on the basis of either id or name.
+            let key1 = _.includes(entrypoints, a.name) ? entrypoints.indexOf(a.name) : entrypoints.indexOf(a.id),
```

Comment:
Time complexity to high here. We are doing an includes and then indexOf. Need some justification for such a slow operation.

Reply:
@saswatds
Ut was required for both id and name lookup.
Could you please guide me regarding the correct way?